salt/grains/core.py

```diff
@@ -1668,6 +1668,28 @@ def append_domain():
         grain['append_domain'] = __opts__['append_domain']
     return grain
 
+def fqdns():
+    '''
+    Return all FQDNs grains for the system
+    '''
+    # Provides:
+    # fqdns
+
+    grains = {}
+    fqdns = set()
+
+    addresses = sum([[_INTERFACES[k]['inet'][0]['address'], 
```

Comment:
I would rather write this as:

```python
addresses_ipv4 = salt.utils.network.ip_addrs()
addresses_ipv6 = salt.utils.network.ip_addrs6()
```
This way we would prevent crashing i.a. in cases where there is no `inet6` address for a given interface.

Reply:
We initially used that approach, but @moio asked to use cached information

Reply:
@mbologna are you sure that information is always present at the time of this call?

Reply:
According to https://github.com/saltstack/salt/blob/2016.11/salt/utils/network.py#L833 it would be possible to reuse the cached `_INTERFACES` when calling `salt.utils.network.ip_addrs()`:

```python
addresses_ipv4 = salt.utils.network.ip_addrs(interface_data=_INTERFACES)
```
I'm in favor of reusing `salt.utils.network.ip_addrs` as it already contains logic to properly extract the IP address and catch possible exceptions.

Reply:
That sounds like the best of the two worlds! :trophy: :earth_africa: :earth_africa: